js/reference.js

```diff
@@ -5165,6 +5239,10 @@
                 // TODO should we move these to the _getFormattedKeyValues?
                 // the multi-fk all-outbounds
                 var sm = this._pageRef.table.sourceDefinitions.sourceMapping;
+
+                // TODO the problem here is that activeList might have been called before, so we don't have the allOutBounds
+                // Currently the wait_for is under the tuple and therefore needs the tuple...
+                // TODO HEEEEEREEEEE!!!!!
```

Comment:
I think you left a TODO in here

Reply:
And this is resolved too by moving the `Citation` object (and therefore it's `waitFor` definition) to the `Reference` API.